graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/fetchers/admin/SingleNamespaceFetcher.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.schemafirst.fetchers.admin;
+
+import graphql.schema.DataFetchingEnvironment;
+import io.stargate.auth.AuthenticationSubject;
+import io.stargate.auth.AuthorizationService;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.db.schema.Keyspace;
+import java.util.Map;
+
+public class SingleNamespaceFetcher extends NamespaceFetcher<Map<String, Object>> {
+
+  public SingleNamespaceFetcher(
+      AuthorizationService authorizationService, DataStoreFactory dataStoreFactory) {
+    super(authorizationService, dataStoreFactory);
+  }
+
+  @Override
+  protected Map<String, Object> get(
+      DataFetchingEnvironment environment,
+      DataStore dataStore,
+      AuthenticationSubject authenticationSubject) {
+
+    String name = environment.getArgument("name");
+    Keyspace keyspace = dataStore.schema().keyspace(name);
+
+    if (keyspace == null || !isReadAuthorized(keyspace, authenticationSubject)) {
+      throw new IllegalArgumentException(String.format("Unknown keyspace '%s'", name));
```

Comment:
Does this error get returned to the end user? If it does should the message be "Unknown namespace"?

Reply:
Yes.